### PR TITLE
Add new way to place tiles

### PR DESCRIPTION
**Pull request by marekventur**
This commit adds a new way to place and rotate tiles
at the same time. It also shows which rotations are possible.

Since people are probably used to the current placement
option I've added this one as a setting and used the 'old' one
as a default.

image

----

small note: While I believe that other players might prefer this mode for normal playing, the primary reason for this PR is that we have started playing JCloisterZone via Zoom. One player starts the game and the other watch and tell them where to place tiles when it's their turn. 

Having a way to do all interaction by simple mouse movements makes it easier when one of the "observers" wants to take over (zoom allows the mouse to be taken over) and the rotation handles make it easier to describe which way around a tile should go.

### CODE REVIEW

File: src/main/java/com/jcloisterzone/ui/grid/layer/LittleBuildingActionLayer.java [hunk 1]
```diff
@@ -146,38 +146,28 @@ public void paint(Graphics2D g2) {
...
+        if (newValue != selected) {
+            selected = newValue;
+            gridPanel.repaint();
+        }
     }
 
     @Override
     public void onShow() {
         super.onShow();
-        //TODO extract listener from this
```
Comment (marekventur) [inline src/main/java/com/jcloisterzone/ui/grid/layer/LittleBuildingActionLayer.java:179]: I've decided to fix this todo instead of copying the hack, hence the unrelated change.

**Comment by farin** [comment]
I like the idea.

Only odd thing is that you can still use tab right mouse to change "default rotation" and maybe this should be respected.

**Comment by farin** [comment]
Anyway I am going to finish reported bug fixes and then I will focus on this PR
Want to merge it and than make new release (probably next week)

**Comment by farin** [comment]
@marekventur I changed your code a little bit (keeping rotation aligned with rotation set through tab). I also remove auto rotation when multiple rotations exists. But I don't strong opinion about it and It's needed to try it more what feels better

You can help trying and testing upcoming release
https://jcloisterzone.com/builds/JCloisterZone-4.6.0-rc.7z
(play server is already updated to be able host 4.6.0 games)
